app/services/idv/financials_step.rb

```diff
@@ -0,0 +1,32 @@
+module Idv
+  class FinancialsStep < Step
+    def complete?
+      idv_form.finance_type && idv_session.financials_confirmation.try(:success?) ? true : false
+    end
```

Comment:
What I don't like about this is that it bases the completeness on side effects. Think of the FinancialsStep class as an operation that performs multiple tasks. In this case, there are 2 main tasks that need to be performed: 

1. validate the form
2. validate the submitted data with the vendor if the form was valid

Then return a result hash of the operation that indicates success or failure.

If everything passed, perform some additional tasks, such as updating the session.